torch/testing/_internal/common_methods_invocations.py

```diff
@@ -290,6 +290,13 @@ def np_sinc_with_fp16_as_fp32(x):
     else:
         return np.sinc(x)
 
+def sample_inputs_tensor_split(op_info, device, dtype, requires_grad):
+    return (SampleInput(make_tensor((S, S, S), device, dtype,
+                                    low=None, high=None,
+                                    requires_grad=requires_grad),
+                        args=(make_tensor((S,), device, torch.int64, low=None, high=None, requires_grad=False),),
```

Comment:
I think this is the cause of the test failures. The splits must be on the CPU, but here they're being put on the passed device. Changing device to be "cpu" should work.

Reply:
There are a few other things to change with this SampleInput:

- add a second sample input where the indices_or_sections is a scalar tensor
- add a third sample input where the dim argument is used
- just create the indices_or_sections tensors directly instead of using `make_tensor`, that is, just call `torch.tensor(...` so the splits make sense given the input tensor. If the splits are randomly generated, for example, they might look very strange, like `(-5, 2, 1, ...)`